sdk/synapse/Azure.Analytics.Synapse.Artifacts/tests/DataFlowDebugSessionClientLiveTests.cs

```diff
@@ -0,0 +1,122 @@
+using NUnit.Framework;
+
+namespace Azure.Analytics.Synapse.Artifacts.Tests
+{
+    /// <summary>
+    /// The suite of tests for the <see cref="DataFlowDebugSessionClient"/> class.
+    /// </summary>
+    /// <remarks>
+    /// These tests have a dependency on live Azure services and may incur costs for the associated
+    /// Azure subscription.
+    /// </remarks>
+    public class DataFlowDebugSessionClientLiveTests : RecordedTestBase<SynapseTestEnvironment>
+    {
+        internal class DisposableDataFlowDebugSession : IAsyncDisposable
+        {
+            private readonly DataFlowDebugSessionClient _client;
+            public string SessionId;
+
+            private DisposableDataFlowDebugSession (DataFlowDebugSessionClient client, string sessionId)
+            {
+                _client = client;
+                SessionId = sessionId;
+            }
+
+            public static async ValueTask<DisposableDataFlowDebugSession> Create (DataFlowDebugSessionClient client, TestRecording recording) =>
+                new DisposableDataFlowDebugSession (client, await CreateResource(client, recording));
+
+            public static async ValueTask<string> CreateResource (DataFlowDebugSessionClient client, TestRecording recording)
+            {
+                // SYNAPSE_API_ISSUE - When do we need to pass DataFlowName?
+                DataFlowDebugSessionCreateDataFlowDebugSessionOperation create = await client.StartCreateDataFlowDebugSessionAsync (new CreateDataFlowDebugSessionRequest ());
+                // SYNAPSE_API_ISSUE - Why is there is no wrapper to save here?
+                return (await create.WaitForCompletionAsync()).Value.SessionId;
+            }
+
+            public async ValueTask DisposeAsync()
+            {
+                // SYNAPSE_API_ISSUE - Why does the ctor not take these? When do we need to pass DataFlowName?
+                DeleteDataFlowDebugSessionRequest deleteRequest = new DeleteDataFlowDebugSessionRequest ()  { SessionId = SessionId };
+                await _client.DeleteDataFlowDebugSessionAsync (deleteRequest);
+            }
+        }
+
+        public DataFlowDebugSessionClientLiveTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        private DataFlowClient CreateFlowClient()
+        {
+            return InstrumentClient(new DataFlowClient(
+                new Uri(TestEnvironment.EndpointUrl),
+                TestEnvironment.Credential,
+                InstrumentClientOptions(new ArtifactsClientOptions())
+            ));
+        }
+
+        private DataFlowDebugSessionClient CreateDebugClient()
+        {
+            return InstrumentClient(new DataFlowDebugSessionClient(
+                new Uri(TestEnvironment.EndpointUrl),
+                TestEnvironment.Credential,
+                InstrumentClientOptions(new ArtifactsClientOptions())
+            ));
+        }
+
+        [Ignore ("SYNAPSE_API_ISSUE issues preventing test from working")]
```

Comment:
Do we have an issue that we could link to here?

Reply:
Let me file one.